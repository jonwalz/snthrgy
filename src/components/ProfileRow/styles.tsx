import styled from "styled-components";
import colors from "../../design/colors";

export const ProfileRowStyles = styled.div`
  border-bottom: 2px solid ${colors.grey};
  padding: 12px 16px;
`;
