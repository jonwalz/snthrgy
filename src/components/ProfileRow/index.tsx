import React, {ReactElement} from "react";
import {useAuth0} from "@auth0/auth0-react";
import {ProfileRowStyles} from "./styles";
import {Button} from "../../components/Button";

interface PropfileRowProps {}

export default function ProfileRow({}: PropfileRowProps): ReactElement {
  const {loginWithPopup, logout, isAuthenticated} = useAuth0();
  const message = isAuthenticated ? "Log out" : "Log in";
  const handler = isAuthenticated ? logout : loginWithPopup;

  return (
    <ProfileRowStyles>
      {/* TODO: this will become a dropdown menu */}
      <Button onClick={handler}>{message}</Button>
    </ProfileRowStyles>
  );
}
