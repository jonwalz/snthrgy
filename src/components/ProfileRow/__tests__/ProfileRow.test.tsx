import React from "react";
import {render, screen} from "@testing-library/react";
import ProfileRow from "../index";

describe("ProfileRow", () => {
  it("should render correctly", () => {
    const {asFragment} = render(<ProfileRow />);

    screen.getByText("Log in");
    expect(asFragment).toMatchSnapshot();
  });
});
