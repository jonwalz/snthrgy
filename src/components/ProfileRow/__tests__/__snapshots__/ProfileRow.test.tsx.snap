// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileRow should render correctly 1`] = `[Function]`;
